web/apps/labelstudio/src/pages/DataManager/DataManager.jsx

```diff
@@ -96,7 +96,27 @@ export const DataManagerPage = ({ ...props }) => {
 
     Object.assign(window, { dataManager });
 
-    dataManager.on("crash", () => setCrashed());
+    dataManager.on("crash", (details) => {
+      const error = details?.error;
+      const isMissingTaskError = error?.startsWith("Task ID:");
+      const isMissingProjectError = error?.startsWith("Project ID:");
+
+      if (isMissingTaskError || isMissingProjectError) {
+        toast.show({
+          message: `The ${
+            isMissingTaskError ? "task" : "project"
+          } you are trying to access does not exist or is no longer available.`,
+          type: ToastType.error,
+          duration: 10000,
+        });
+      }
+
+      if (isMissingTaskError) {
+        history.push(buildLink("/", { id: params.id }));
```

Comment:
Current code will redirect to pages like `/projects/123/` with trailing slash which we normally don't use
```suggestion
        history.push(buildLink("", { id: params.id }));
```